### PR TITLE
Screens on activity completion & API integration with themes

### PR DESCRIPTION
**Pull request by BamMironov**

### CODE REVIEW

File: src/shared/api/services/answerService.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  activityId: string;
+  answers: AnswerDto[];
+};
+
+type ActivityAnswersResponse = SuccessfulEmptyResponse;
+
+type ActivityFlowAnswersRequest = {
+  appletId: string;
+  appletHistoryVersion: string;
+  flowItemHistoryId: string;
```
Comment (anq83) [inline src/shared/api/services/answerService.ts:23]: It should be discussed, it is not clear about history properties here

File: src/features/pass-survey/model/hooks/useActivityState.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  );
+
+  const shouldCreateStorage = !activityStorage && pipeline.length;
+
+  if (shouldCreateStorage) {
+    createActivityStorage();
+  }
+
+  function createActivityStorage() {
+    updateActivityStorage({
```
Comment (anq83) [inline src/features/pass-survey/model/hooks/useActivityState.ts:81]: Maybe upsertActivityStorage = update / insert  - often used on backend for similar cases

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/entities/activity/api/useActivityAnswersMutation.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  useBaseMutation,
+  AnswerService,
+  MutationOptions,
+} from '@app/shared/api';
+
+type Options = MutationOptions<typeof AnswerService.sendActivityAnswers>;
+
+export const useActivityAnswersMutation = (options?: Options) => {
+  return useBaseMutation(
+    ['activity-answers'],
```
[the remark this answers is not in the document]
Comment (BamMironov) [reply src/entities/activity/api/useActivityAnswersMutation.ts:11]: Correct. Will fix it

[the remark this answers is not in the document]
Comment (BamMironov) [reply src/entities/activity/api/useActivityAnswersMutation.ts:11]: I have decided to get rid of mutation keys from all the mutations. They are useless in our case.

File: src/entities/activity/model/mappers.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+} from '../lib';
+
+export function mapToActivity(dto: ActivityDto): ActivityDetails {
+  return {
+    ...dto,
+    items: dto.items.map(item => ({
+      ...item,
+      inputType: item.inputType as ActivityItemType,
+      config: item.config as ActivityItemConfig,
+    })) as ActivityItem[],
```
[the remark this answers is not in the document]
Comment (BamMironov) [reply src/entities/activity/model/mappers.ts:17]: Yes

File: src/shared/api/services/answerService.ts (hunk 1 above)
Comment (BamMironov) [reply src/shared/api/services/answerService.ts:23]: Yes, we can talk about it during the next architecture meeting

The endpoint: 
https://api-dev.cmiml.net/docs#/Answers/answer_activity_item_create_answers_activity_items_post

**Pull request merged**